packages/components/src/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, ReactNode } from 'react';
+import React from 'react';
+import { type DropdownProps } from "@/dropdown/Dropdown.types";
+import { globalStyles } from "@/index.styled";
+import { DropdownContainer, DropdownLabel, DropdownChildren, DropdownItem, darkTheme } from "@/dropdown/Dropdown.styled"; 
+import chevronDown from "@/assets/chevron-down.svg"
+
+export const Dropdown = ({
+        label = "Select...", 
+        theme, 
+        showingOptions: initialShowingOptions, 
+        disabled, 
+        error, 
+        children}: DropdownProps) => {
+
+    globalStyles();
+
+    const [showingOptions, setShowingOptions] = useState(initialShowingOptions);
+    const [selectedOption, setSelectedOption] = useState('');
+
+    const toggleDropdownOptions = () => {
+        if(disabled) {
+            setShowingOptions(false);
+            return;
+        };
+        setShowingOptions(!showingOptions);
+    }
+
+    const handleDropdownKeyDown = (event: React.KeyboardEvent) => {
+        if (event.key === 'Enter' || event.key === ' ') {
+          toggleDropdownOptions();
+        }
+    }
+
+    const handleOptionClick = (option: string) => {
+        setSelectedOption(option);
+        setShowingOptions(false);
+        console.log(`Selected option: ${option}`);
+    }
+
+    const handleDropdownOptionKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter' || event.key === ' ') {
+        if (event.currentTarget.textContent) {
+            handleOptionClick(event.currentTarget.textContent);
+        }
+    }
+}
+  
+    return (
+        <DropdownContainer
```

Comment:
Just a suggestion, but using the <select> and <option> tag instead of a normal <div> tag works better for accessibility. This eliminates the need to implement the keyboard event as well.

Reply:
Hi, yes I did do that initially. But the `<option>` had limited customization and mostly relies on the browser of the user.